fourier.c

```diff
@@ -434,6 +454,16 @@ fourier_create_procedure(GimpPlugIn *plug_in,
     gimp_procedure_set_sensitivity_mask(procedure,
                                         GIMP_PROCEDURE_SENSITIVE_DRAWABLE);
 
+#ifdef HAVE_GETTEXT
```

Comment:
This block is included 3 times ; is it required or maybe because of the multiples commits?

Reply:
Yes. Needed three times.
The plugins included with gimp3 rely on the po locales provided by gimp3. We do not, and cannot have access to using those po locales, so we need to follow the usual route of providing your own po locales (which are in subdirectory po).
The first one at the top is a switch.
If building with `gimptool-3.0` we turn po locales off. This makes it so fourier3.exe does not use locales.... in linux, fourier would be installed in ~/.local and you have no access to root. With windows, it is also a local install.
So far we don't know what gimp-3.0 behaves like, since it is still set as gimp-2.99 in gimp rc0.
If building with ./configure,make, then we know you are going to install as root, and therefore can install the po locales too.
The second one, is when setting up procedures, you'll need to provide local support in that function.
The third one, is when running the main program, you'll need to provide local support there.

Reply:
Yes I understand the 3 different textdomain loading in 1 block of #ifdef...#endif.  I was mentioning that this block is duplicated 3 times, in lines 410, 457 and 639. Using this block only once in the fourier_init function would not be enough?

Reply:
410 needs to be there because fourier_create_procedure() is part of a linked list belonging to the menu. It could be at 401, but the first place needed is 420 to do GETTEXT() conversion of _("_Fourier..."), and the following variables.
This only happens if (name==PLUG_IN_PROC)

fourier_create_procedure() also needs to setup for your 2nd program PLUG_IN_DIR_PROC at line 446, so once again you need GETTEXT() to translate _(PLUG_IN_DIR_MENU_LABEL).
This only happens if (name==PLUG_IN_DIR_PROC)

If (name==something_else) go use gimp po files, or another po files belonging to a different project.

The above I believe is part of a linked list belonging to the pull-down menus and also the plug-in browser help menus.

Line 639 is where your program actually runs what it needs to do as part of fourier_run(), and the first place it might be needed is at line 656 if there is an error there, otherwise it will likely be seen a little later when you call plugin_dialog().

Reply:
Coming to think about it, I don't know if fourier_init() would work ? mabe it does. Have not tried that.